test/exe/Main.hs

```diff
@@ -1144,6 +1145,40 @@ insertNewDefinitionTests = testGroup "insert new definition actions"
         ++ txtB')
   ]
 
+
+deleteUnusedDefinitionTests :: TestTree
+deleteUnusedDefinitionTests = testGroup "delete unused definition action"
+  [ testSession "delete unused top level binding" $ do
+      let source = T.unlines [ "{-# OPTIONS_GHC -Wunused-top-binds #-}"
+                             , "module A (some) where"
+                             , ""
+                             , "f :: Int -> Int"
+                             , "f 1 = let a = 1"
+                             , "      in a"
+                             , "f 2 = 2"
+                             , ""
+                             , "some = ()"
+                             ]
+      docId <- createDoc "A.hs" "haskell" source
+      expectDiagnostics [ ("A.hs", [(DsWarning, (4, 0), "not used")]) ]
+
+      Just (CACodeAction action@CodeAction { _title = actionTitle })
+                  <- find (\(CACodeAction CodeAction{_title=x}) -> "Delete" `T.isPrefixOf` x )
+                     <$> getCodeActions docId (R 0 0 0 0)
+
+      liftIO $ actionTitle @?= "Delete ‘f’"
+      executeCodeAction action
+      contentAfterAction <- documentContents docId
+      liftIO $ contentAfterAction @?= T.unlines [
+        "{-# OPTIONS_GHC -Wunused-top-binds #-}"
+        , "module A (some) where"
+        , ""
+        , ""
```

Comment:
A very minor remark, but beforehand they had 1-line gaps, now they have 2-line gaps. Should we expand one line after if it's blank? Or maybe this is getting ahead of ourselves and there is no need.

Reply:
Makes sense to me. What do you think @serhiip?

Reply:
For sure expanding the text edit until the next binding in a file is a good thing. I was thinking about this initially but only after your comment I've figured how to actually do it: now the code works by finding the next binding after current one and expands the deletion until that next binding (if any). To make that happen I rearranged my code a bit. Hope it makes more sense now. Thanks for suggestion